cvasl/harmonizer_cli.py

```diff
@@ -0,0 +1,225 @@
+import argparse
+import pandas as pd
+import sys
+import os
+from cvasl.dataset import *
+from cvasl.harmonizers import *
+from cvasl.prediction import *
+
+def main():
+    parser = argparse.ArgumentParser(description="Harmonize MRI datasets using various methods.")
+
+    # Data parameters
+    parser.add_argument("--dataset_paths", required=True, help="Comma-separated paths to dataset CSV files. For datasets with multiple paths, use semicolon to separate paths, e.g. 'path1;path2',path3,path4")
+    parser.add_argument("--site_ids", required=True, help="Comma-separated site IDs for each dataset.")
+    parser.add_argument("--patient_identifier", default="participant_id", help="Column name for patient identifier.")
+    parser.add_argument("--features_to_drop", default="m0,id", help="Comma-separated list of features to drop.")
+    parser.add_argument("--features_to_map", default="readout,labelling,sex", help="Comma-separated list of categorical features to encode.")
+    parser.add_argument("--decade", type=str, default="True", help="Whether to add decade-related features (True/False).")
+    parser.add_argument("--icv", type=str, default="True", help="Whether to add ICV-related features (True/False).")
+
+    # Harmonization method and parameters
+    parser.add_argument("--method", required=True, choices=["neuroharmonize", "covbat", "neurocombat", "nestedcombat", "comscanneuroharmonize", "autocombat", "relief", "combat++"], help="Harmonization method to use.")
+    
+    # NeuroHarmonize parameters
+    parser.add_argument("--nh_features_to_harmonize", default="aca_b_cov,mca_b_cov,pca_b_cov,totalgm_b_cov,aca_b_cbf,mca_b_cbf,pca_b_cbf,totalgm_b_cbf", help="Comma-separated features to harmonize (for NeuroHarmonize).")
```

Comment:
I'm not sure it makes sense to specify the parameters as cli arguments for each method. In this case, I'd put these in a config file and have the user specify the path of the config file.

The main reason is that I expect the call to become too long (read messy) and no user will use it that way.